Classes/Service/CompressImageService.php

```diff
@@ -172,7 +173,28 @@ public function initializeCompression($file): void
         } else {
             $this->addMessageToFlashMessageQueue('debugMode', [], FlashMessage::INFO);
         }
+    }
+
+    /**
+     * @param File $file
+     * @param ProcessedFile $processedFile
+     */
+    public function compressProcessedFile(File $file, ProcessedFile $processedFile): void
+    {
+        $this->initAction();
+        if ($this->isFileInExcludeFolder($file)) {
+            return;
+        }
+        if (!in_array(strtolower($file->getMimeType()), ['image/png', 'image/jpeg'], true)) {
+            return;
+        }
+        if ((int)$this->settings['debug'] !== 0) {
```

Comment:
can you elaborate on this setting? "if debug is enabled, don't do anything"?

if so, you can move this option to the very first setting to avoid other checks (for perf reasons)